hardhat.config.ts

```diff
@@ -41,6 +41,71 @@ const config: HardhatUserConfig = {
         version: "0.8.27",
       },
     ],
+    overrides: {
+      "contracts/verifiers/UniversalVerifier.sol": {
+        version: "0.8.27",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+      "contracts/test-helpers/VerifierTestWrapper.sol": {
+        version: "0.8.27",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+      "contracts/test-helpers/EmbeddedVerifierWrapper.sol": {
+        version: "0.8.27",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+      "contracts/test-helpers/RequestDisableableTestWrapper.sol": {
+        version: "0.8.27",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+      "contracts/test-helpers/RequestOwnershipTestWrapper.sol": {
+        version: "0.8.27",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+      "contracts/test-helpers/ValidatorWhitelistTestWrapper.sol": {
+        version: "0.8.27",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+      "contracts/state/State.sol": {
+        version: "0.8.27",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
```

Comment:
are we sure that current production version has '200' runs and not '1000' , won't it influence the upgrade

Reply:
Currently we were not working with optimizer in the deployments we have. We achieved max limit size of the contract bytecode and with this specific settings for every contract that is at maximum limit size it reduce this in most cases to half of the size and we are able to deploy them.